biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java

```diff
@@ -0,0 +1,1394 @@
+
+
+        }
+        if (debug)
+            System.out.println("length exons: " + lengthExons + " codin length: " + (codingLength - 3));
+        return codingLength - 3;
+    }
+
+    /**
+     * Get the length of the coding sequence
+     *
+     * @param exonStarts
+     * @param exonEnds
+     * @param cdsStart
+     * @param cdsEnd
+     * @return
+     */
+    public static int getCDSLengthForward(List<Integer> exonStarts, List<Integer> exonEnds,
+                                          int cdsStart, int cdsEnd) {
+        boolean inCoding = false;
+        int codingLength = 0;
+
+        int lengthExons = 0;
+        // map forward
+        for (int i = 0; i < exonStarts.size(); i++) {
+
+            int start = exonStarts.get(i);
+            int end = exonEnds.get(i);
+            lengthExons += end - start;
+
+            if (debug)
+                System.out.println("forward exon: " + (start+1) + " - " + end + " | " + (end - start));
+
+            if (start+1 <= cdsStart +1 && end >= cdsStart+1) {
+
+                inCoding = true;
+                codingLength += (end - cdsStart);
+                if (debug) {
+                    System.out.println("     UTR         : " + start + " - " + (cdsStart ));
+                    System.out.println(" ->  Exon        : " + (cdsStart+1) + " - " + end + " | " + (end - cdsStart+1) + " | " + codingLength + " | " + (codingLength % 3));
+                }
+            } else if (start+1 <= cdsEnd && end >= cdsEnd) {
+                //System.out.println(" <-- CDS end at: " + cdsEnd );
+                inCoding = false;
+                codingLength += (cdsEnd - start);
+                if (debug) {
+                    System.out.println(" <-  Exon        : " + (start +1)+ " - " + cdsEnd + " | " + (cdsEnd - start+1) + " | " + codingLength + " | " + (codingLength % 3));
+                    System.out.println("     UTR         : " + cdsEnd + 1 + " - " + end);
+                }
+
+            } else if (inCoding) {
+                // full exon is coding
+                codingLength += (end - start);
+                if (debug)
+                    System.out.println("     Exon        :" + (start+1) + " - " + end + " | " + (end - start+1) + " | " + codingLength + " | " + (codingLength % 3));
+            }
+            //			if ( inCoding )
+            //				System.out.println("exon phase at end:" + (codingLength % 3));
+            //
+            //			System.out.println("   coding length: " + codingLength);
+
+
+        }
+        if (debug) {
+            System.out.println("length exons: " + lengthExons);
+            System.out.println("CDS length:" + (codingLength-3));
+        }
+        return codingLength-3 ;
+    }
+
+
+
+    /** Extracts the exon boundaries in CDS coordinates. (needs to be divided by 3 to get AA positions)
+     *
+     * @param chromPos
+     * @return
+     */
+    public static List<Range> getCDSExonRanges(GeneChromosomePosition chromPos){
```

Comment:
Range should be parameterised, the compiler gives warnings for it. Same for all other occurrences below